cmd/kubeadm/app/cmd/config.go

```diff
@@ -272,6 +273,48 @@ func newCmdConfigMigrate(out io.Writer) *cobra.Command {
 	return cmd
 }
 
+// newCmdConfigValidate returns cobra.Command for the "kubeadm config validate" command
+func newCmdConfigValidate() *cobra.Command {
+	var cfgPath string
+
+	cmd := &cobra.Command{
+		Use:   "validate",
+		Short: "Read a file containing the kubeadm configuration API and report any validation problems",
+		Long: fmt.Sprintf(dedent.Dedent(`
+			This command lets you validate a kubeadm configuration API file and report any warnings and errors.
+			If there are no errors the exit status will be zero, otherwise it will be non-zero.
+			Any unmarshaling problems such as unknown API fields will trigger warnings. Unknown API versions and
+			fields with invalid values will trigger fatal errors. Any other errors or warnings may be reported
+			depending on contents of the input file.
+
+			In this version of kubeadm, the following API versions are supported:
+			- %s
+		`), kubeadmapiv1.SchemeGroupVersion),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(cfgPath) == 0 {
+				return errors.Errorf("the --%s flag is mandatory", options.CfgPath)
+			}
+
+			cfgBytes, err := os.ReadFile(cfgPath)
+			if err != nil {
+				return err
+			}
+
+			// MigrateOldConfig performs conversion to internal type, strict unmarshaling,
+			// defaulting and validation. It can be used for the validation purpose with output bytes
+			// ignored.
+			if _, err = configutil.MigrateOldConfig(cfgBytes); err != nil {
+				return err
+			}
```

Comment:
I'd prefer to print something if the validation is okay, nothing output to stdout now if everything is good.

Reply:
perhaps, we should. i'd like to get more opinions on this. the exit code 0 / 1 may be sufficient.

Reply:
looks like there are no comments on this. my preference is to not print on stdout on "OK", similarly to some go tools.

Reply:
If making it like the dry-run, we can print a result for each configuration in the file
```
[root@daocloud ~]# ./kubeadm-dev  config  validate --config kubeadmcfg.yaml
InitConfiguration validated
ClusterConfiguration validated
Bar unknown
```
This would be much clear if there are multi items in the configuration files.


Or, we can just print `OK`/`Validated`/`Done` when there is no error/warning. This is also good.

Reply:
actually, there is no clean way to detect if a warning was printed. it just goes to stderr from klog. non warnings go there too.

printing per validated config is possible but it will complicate the plumbing of functions a bit. we don't want to print the same messages during normal init/join...but the backend functions are the same.

i will check what options witk well.

Reply:
updated to just print "ok" if validation has no errors.